app/routes.py

```diff
@@ -0,0 +1,306 @@
+from .__init__ import *
+
+
+from .MyFunctions import *
+# from .models import *
+from .schemas import *
+from .sockets import *
+
+
+# user signup
+@app.route('/user/signup', methods=['POST'])
+def user_signup():
+
+    username=request.json['username']
+    password = request.json['password']
+    email=request.json['email']
+
+    email_check = Users.query.filter_by(email=email).first()
+
+    if email_check:
+        return jsonify({
+            'username':'',
+            'password':'',
+            'email':'',
+            'admin_status':0,
+            'jwt':'',
+            'status':'User Already Exsists'
+        })
+
+    new_user = Users(username, password, email)
+    db.session.add(new_user)
+    db.session.commit()
+
+    user = Users.query.filter_by(email=request.json['email']).first()
+    auth_token = encode_auth_token(user.id)
+
+    try:
+        TokenValue = auth_token.decode()
+    except: # TypeError object has no attribute Token
+        return jsonify({
+        'username':username,
+        'password':password,
+        'email':email,
+        'admin_status':0,
+        'jwt':'',
+        'status':"Failed to generate Token"
+        })
+
+    return jsonify({
+        'username':username,
+        'password':password,
+        'email':email,
+        'admin_status':0,
+        'jwt':TokenValue,
+        'status':"User Created Successfully"
+    })
+
+
+# user login
+@app.route('/user/login', methods=['POST'])
+def user_login():
+
+    email=request.json['email']
+    password=request.json['password']
+    
+    try:
+        user = Users.query.filter_by(email=request.json['email']).first()
+
+        if password == user.password:
+            auth_token = encode_auth_token(user.id)
+            return jsonify({
+                'status':'success',
+                'message':'Successfully logged in',
+                'auth_token':auth_token.decode()
+            })
```

Comment:
Add else part here because when the user will directly login without signup then he will get an error wrong creds. But he should get user does not exists. In exception you can add user does not exists and in the if part add wrong credentials.

Reply:
@Geek-ubaid will make this change.

Reply:
dealing a major bug here:
when user enters wrong password
It is supposed to be handled properly but it trhows error 
    "The view function did not return a valid response. The"
TypeError: The view function did not return a valid response. The function either returned None or ended without a return statement.

Traceback is not telling from where it's originating from my code, and even debug statements failed to help. Working on this.

Reply:
@Geek-ubaid 
[Resolved](https://github.com/GDGVIT/AttendanceApp-Backend/pull/8/commits/c9b874d339e6b4149eb4714058f5ea59f47b9dbf)